// @coreui/angular/esm2022/lib/utilities/text-bg-color.directive.mjs
import { Directive, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class TextBgColorDirective {
    constructor() {
        /**
         * Set text-bg-color of element
         * @type Colors
         */
        this.textBgColor = '';
    }
    get hostClasses() {
        const color = this.textBgColor;
        return {
            [`text-bg-${color}`]: !!color
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.4", ngImport: i0, type: TextBgColorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.3.4", type: TextBgColorDirective, isStandalone: true, selector: "[cTextBgColor]", inputs: { textBgColor: ["cTextBgColor", "textBgColor"] }, host: { properties: { "class": "this.hostClasses" } }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.4", ngImport: i0, type: TextBgColorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[cTextBgColor]',
                    standalone: true
                }]
        }], propDecorators: { textBgColor: [{
                type: Input,
                args: ['cTextBgColor']
            }], hostClasses: [{
                type: HostBinding,
                args: ['class']
            }] } });
